var/spack/repos/builtin/packages/geant4/package.py

```diff
@@ -51,7 +49,8 @@ class Geant4(CMakePackage):
     depends_on("xerces-c cxxstd=17", when="cxxstd=17")
     depends_on("clhep@2.3.3.0: cxxstd=17", when="@10.03.p03: cxxstd=17")
     patch('cxx17.patch', when='@:10.03.p99 cxxstd=17')
-    patch('cxx17_geant4_10_0.patch', level=1, when='@10.04.00: cxxstd=17')
+    patch('cxx17_geant4_10_0.patch', level=1, when='@10.04.00:10.04.99 cxxstd=17')
+    patch('geant4-10.05.p01-Wno_register.patch', level=1, when='@10.05.p01')
```

Comment:
Why was this patch actually needed?

Reply:
`register` keyword is deprecated in C++17. This produces a warning, which is treated as an error.